docs/introduction.md

```diff
@@ -0,0 +1,125 @@
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional webNote iosNote androidNote" />
+
+```SnackPlayer name=Hello%20World
+import React from 'react';
+import { Text, View } from 'react-native';
+
+export default function YourApp() {
+  return (
+      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+        <Text>Hello, world!</Text>
+      </View>
+  );
+}
+```
+
+<block class="classical webNote iosNote androidNote" />
+
+```SnackPlayer name=Hello%20World
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+
+export default class HelloWorldApp extends Component {
+  render() {
+    return (
+      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+        <Text>Hello, world!</Text>
+      </View>
+    );
+  }
+}
+
+```
+
+<block class="classical functional webNote iosNote androidNote" />
+
+The above is a Snack Player. It’s [a handy tool created by Expo](https://expo.io/tools#snack) to embed and run React Native projects on and share how they render in platforms like Android and iOS. We will be using them a lot to demonstrate code samples throughout these docs. You can play around with sample code provided in the Snack, or you can [set up an environment on your local machine](getting-started) and paste the code into your `App.js` file there.
+
+## Function and Class Syntaxes
+
+You may have noticed the two tabs for “Functional Syntax” and “Class Syntax” in the Snack above. We thought it would be useful for you to have both syntaxes to study, though we will only go into depth with function components.
+
+Classes were the only way to manage state in a React component until [React 16.8 introduced an API called “Hooks” for managing state and more in function components](https://reactjs.org/docs/hooks-intro.html). In March 2019, [React Native 0.58 adopted Hooks as well](https://facebook.github.io/react-native/blog/2019/03/12/releasing-react-native-059). While Hooks are the future-facing way to write your React-based components, you might be working in a codebase that still uses classes.
+
+With function components, you export a JavaScript function that returns your component(s):
+
+```jsx
+export default function YourApp() {
+  return (
+    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
+      <Text>Hello, world!</Text>
+    </View>
+  );
+}
+```
+
+> The sample code in this documentation uses [JavaScript version ES2015](https://babeljs.io/learn-es2015/) (also known as ES6). Since ES2015 is not supported by all JavaScript engines, React Native ships with ES2015 support out of the box thanks to [Babel](https://babeljs.io/) (which also handles your JSX!). This means you can use ES2015 without worrying about backwards compatibility.
```

Comment:
Could you remove this? All engines that RN runs on since 0.59 are using modern JavaScript (ES2015+) and as such Babel is not required to ship modern features, even though the transform is still enabled by default.

(cc @ide – hey, should we do anything about that? We should be able to disable almost all babel transforms in RN by default but we'd need to figure out a migration strategy. Good app size and possibly perf impact to be had.)

Reply:
Good to know, will do!